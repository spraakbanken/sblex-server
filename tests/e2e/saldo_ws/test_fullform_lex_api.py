import pytest
from fastapi import status
from httpx import AsyncClient


class TestFullformLexRoutes:
    @pytest.mark.parametrize("segment", ["dväljs"])
    @pytest.mark.asyncio
    async def test_json_valid_input_returns_200(
        self, client: AsyncClient, segment: str, snapshot_json
    ) -> None:
        res = await client.get(f"/fl/json/{segment}")
        assert res.status_code == status.HTTP_200_OK
        assert res.json() == snapshot_json

    @pytest.mark.parametrize("segment", ["tt"])
    @pytest.mark.asyncio
    async def test_json_missing_segment_returns_404(
        self, client: AsyncClient, segment: str, snapshot_json
    ) -> None:
        res = await client.get(f"/fl/json/{segment}")
        assert res.status_code == status.HTTP_404_NOT_FOUND
        assert res.json() == snapshot_json

    @pytest.mark.parametrize("segment", ["t", "dväljs"])
    @pytest.mark.asyncio
    async def test_xml_valid_input_returns_200(
        self, client: AsyncClient, segment: str, snapshot
    ) -> None:
        res = await client.get(f"/fl/xml/{segment}")
        assert res.status_code == status.HTTP_200_OK
        assert res.headers["content-type"] == "application/xml"
        assert res.text == snapshot

    @pytest.mark.parametrize("segment", ["", " ", "dväljas", "femrumslägenhet"])
    @pytest.mark.asyncio
    async def test_html_valid_input_returns_200(
        self, client: AsyncClient, segment: str, snapshot
    ) -> None:
        res = await client.get(f"/fl/html?segment={segment}")
        assert res.status_code == status.HTTP_200_OK
        assert res.headers["content-type"] == "text/html; charset=utf-8"
        assert res.text == snapshot

    @pytest.mark.parametrize("segment", ["löparsko"])
    @pytest.mark.asyncio
    async def test_html_missing_input_returns_404(
        self, client: AsyncClient, segment: str, snapshot
    ) -> None:
        res = await client.get(f"/fl/html?segment={segment}")
        assert res.status_code == status.HTTP_404_NOT_FOUND
        assert res.headers["content-type"] == "text/html; charset=utf-8"
        assert res.text == snapshot

    @pytest.mark.parametrize("segment", ["", " ", "dväljas", "dväljsxdf"])
    @pytest.mark.asyncio
    async def test_html_orig_valid_input_returns_307(
        self,
        client: AsyncClient,
        segment: str,
    ) -> None:
        res = await client.get(f"/fl/html/{segment}")
        assert res.status_code == status.HTTP_307_TEMPORARY_REDIRECT
