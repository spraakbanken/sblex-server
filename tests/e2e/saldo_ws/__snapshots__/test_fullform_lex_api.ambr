# serializer version: 1
# name: TestFullformLexRoutes.test_html_valid_input_returns_200[ ]
  '''
  <!DOCTYPE html>
  <html>
  
  <head>
    <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
    <link rel="shortcut icon" href="https://svn.spraakdata.gu.se/sb-arkiv/pub/images/saldo_icon.png">
    <link rel="stylesheet" type="text/css" href="http://testserver/static/saldo.css" />
    <title></title>
    
    <script src="https://unpkg.com/htmx.org@1.9.8" crossorigin="anonymous"></script>
  </head>
  
  
    <body>
      
      <center>
        <p>
          <a href="https://spraakbanken.gu.se/saldo"><img
              src="https://spraakbanken.gu.se/sites/spraakbanken.gu.se/files/img/saldo/saldo.gif" align="top"
              alt="SALDO" /></a>
        </p>
  
        <script>
          function input_handler(e) {
            var word = document.getElementById('input').value;
            if (word.length > 0) {
              location.href = 'https://spraakbanken.gu.se/ws/saldo-ws//html/' + encodeURIComponent(word);
            }
          }
  
        </script>
        <p><a href="https://svn.spraakdata.gu.se/repos/sblex/pub/saldo_instruktion.pdf">dokumentation (pdf)</a></p>
        
      </center>
  
      <div id="output_table">
        
  <center>
      <form action="http://testserver/fl/html" method="get" class="inputclass">
          <input type="search" id="search" class="inputclass" value="" name="segment" size="30"
              placeholder="Skriv in en ordform" results="10" />
          <input type="submit" value="Sök" />
      </form>
      
  </center>
  
      </div>
    </body>
  
  </html>
  '''
# ---
# name: TestFullformLexRoutes.test_html_valid_input_returns_200[]
  '''
  <!DOCTYPE html>
  <html>
  
  <head>
    <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
    <link rel="shortcut icon" href="https://svn.spraakdata.gu.se/sb-arkiv/pub/images/saldo_icon.png">
    <link rel="stylesheet" type="text/css" href="http://testserver/static/saldo.css" />
    <title></title>
    
    <script src="https://unpkg.com/htmx.org@1.9.8" crossorigin="anonymous"></script>
  </head>
  
  
    <body>
      
      <center>
        <p>
          <a href="https://spraakbanken.gu.se/saldo"><img
              src="https://spraakbanken.gu.se/sites/spraakbanken.gu.se/files/img/saldo/saldo.gif" align="top"
              alt="SALDO" /></a>
        </p>
  
        <script>
          function input_handler(e) {
            var word = document.getElementById('input').value;
            if (word.length > 0) {
              location.href = 'https://spraakbanken.gu.se/ws/saldo-ws//html/' + encodeURIComponent(word);
            }
          }
  
        </script>
        <p><a href="https://svn.spraakdata.gu.se/repos/sblex/pub/saldo_instruktion.pdf">dokumentation (pdf)</a></p>
        
      </center>
  
      <div id="output_table">
        
  <center>
      <form action="http://testserver/fl/html" method="get" class="inputclass">
          <input type="search" id="search" class="inputclass" value="" name="segment" size="30"
              placeholder="Skriv in en ordform" results="10" />
          <input type="submit" value="Sök" />
      </form>
      
  </center>
  
      </div>
    </body>
  
  </html>
  '''
# ---
# name: TestFullformLexRoutes.test_html_valid_input_returns_200[dv\xe4ljas]
  '''
  <!DOCTYPE html>
  <html>
  
  <head>
    <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
    <link rel="shortcut icon" href="https://svn.spraakdata.gu.se/sb-arkiv/pub/images/saldo_icon.png">
    <link rel="stylesheet" type="text/css" href="http://testserver/static/saldo.css" />
    <title>dväljas</title>
    
    <script src="https://unpkg.com/htmx.org@1.9.8" crossorigin="anonymous"></script>
  </head>
  
  
    <body>
      
      <center>
        <p>
          <a href="https://spraakbanken.gu.se/saldo"><img
              src="https://spraakbanken.gu.se/sites/spraakbanken.gu.se/files/img/saldo/saldo.gif" align="top"
              alt="SALDO" /></a>
        </p>
  
        <script>
          function input_handler(e) {
            var word = document.getElementById('input').value;
            if (word.length > 0) {
              location.href = 'https://spraakbanken.gu.se/ws/saldo-ws//html/' + encodeURIComponent(word);
            }
          }
  
        </script>
        <p><a href="https://svn.spraakdata.gu.se/repos/sblex/pub/saldo_instruktion.pdf">dokumentation (pdf)</a></p>
        
      </center>
  
      <div id="output_table">
        
  <center>
      <form action="http://testserver/fl/html" method="get" class="inputclass">
          <input type="search" id="search" class="inputclass" value="dväljas" name="segment" size="30"
              placeholder="Skriv in en ordform" results="10" />
          <input type="submit" value="Sök" />
      </form>
      
      <h1>segment: dväljas</h1>
      <p><a href="http://testserver/sms/html/dväljas">sammansättningsanalys (FUNGERAR INTE)</a></p>
      <table border="1">
          
          
          <tr>
              <td>
                  
                  <a href="http://testserver/lid/html/dväljas..1">dväljas..1</a>
                  
              </td>
              <td>
                  
                  <a href="http://testserver/lid/html/bo..1">bo..1</a>
                  
                  
              </td>
              <td><a href="http://testserver/gen/html/vb_vs_dväljas/dväljas">dväljas..vb.1</a>
  
              </td>
              <td><a href="https://spraakbanken.gu.se/korp/#?search=lemgram|dväljas..vb.1">korpus</a></td>
          </tr>
          
          
      </table>
      
  </center>
  
      </div>
    </body>
  
  </html>
  '''
# ---
# name: TestFullformLexRoutes.test_html_valid_input_returns_200[dv\xe4ljsxdf]
  '''
  <!DOCTYPE html>
  <html>
  
  <head>
    <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
    <link rel="shortcut icon" href="https://svn.spraakdata.gu.se/sb-arkiv/pub/images/saldo_icon.png">
    <link rel="stylesheet" type="text/css" href="http://testserver/static/saldo.css" />
    <title>dväljsxdf</title>
    
    <script src="https://unpkg.com/htmx.org@1.9.8" crossorigin="anonymous"></script>
  </head>
  
  
    <body>
      
      <center>
        <p>
          <a href="https://spraakbanken.gu.se/saldo"><img
              src="https://spraakbanken.gu.se/sites/spraakbanken.gu.se/files/img/saldo/saldo.gif" align="top"
              alt="SALDO" /></a>
        </p>
  
        <script>
          function input_handler(e) {
            var word = document.getElementById('input').value;
            if (word.length > 0) {
              location.href = 'https://spraakbanken.gu.se/ws/saldo-ws//html/' + encodeURIComponent(word);
            }
          }
  
        </script>
        <p><a href="https://svn.spraakdata.gu.se/repos/sblex/pub/saldo_instruktion.pdf">dokumentation (pdf)</a></p>
        
      </center>
  
      <div id="output_table">
        
  <center>
      <form action="http://testserver/fl/html" method="get" class="inputclass">
          <input type="search" id="search" class="inputclass" value="dväljsxdf" name="segment" size="30"
              placeholder="Skriv in en ordform" results="10" />
          <input type="submit" value="Sök" />
      </form>
      
      <h1>segment: dväljsxdf</h1>
      <p><a href="http://testserver/sms/html/dväljsxdf">sammansättningsanalys (FUNGERAR INTE)</a></p>
      <table border="1">
          
          <tr>
              <td>ordet saknas i lexikonet</td>
          </tr>
          
      </table>
      
  </center>
  
      </div>
    </body>
  
  </html>
  '''
# ---
# name: TestFullformLexRoutes.test_xml_valid_input_returns_200[dv\xe4ljs]
  '''
  <?xml version="1.0" encoding="UTF-8"?>
  <result>
  <a><id>dväljas..1</id><fm>bo..1</fm><fp>PRIM..1</fp><l>dväljas..vb.1</l><gf>dväljas</gf><p>vb_vs_dväljas</p></a>
  </result>
  '''
# ---
# name: TestFullformLexRoutes.test_xml_valid_input_returns_200[t]
  '''
  <?xml version="1.0" encoding="UTF-8"?>
  <result>
  
  </result>
  '''
# ---
